types/index.d.ts

```diff
@@ -0,0 +1,45 @@
+// Minimum TypeScript Version: 3.2
```

Comment:
Running `dtslint` showed the following error

```
ERROR: 13:14  expect  Compile error in typescript@3.1 but not in typescript@3.2.
Fix with a comment '// Minimum TypeScript Version: 3.2' just under the header.
Argument of type '(settings?: TOCSettings | undefined) => (node: Node) => void' is not assignable to parameter of type 'Attacher<[{ maxDepth: number; }], Partial<RemarkOptions>>'.
  Types of parameters 'settings' and 'settings_0' are incompatible.
    Type '{ maxDepth: number; }' is not assignable to type 'TOCSettings'.
      Types of property 'maxDepth' are incompatible.
        Type 'number' is not assignable to type '1 | 2 | 3 | 4 | 5 | 6 | undefined'.
```

Reply:
Yep, most packages need a min version, for example https://github.com/syntax-tree/mdast-util-toc/blob/a6361867d908bdac1db70baaff566b3fee733a96/types/index.d.ts#L1

Reply:
I started with `// TypeScript Version: 3.0` so was a bit surprised when it suggested using 3.2

Would `mdast-util-toc` have the same issue?

Reply:
Possibly?
I it may be a bug in the type system itself.
https://github.com/remarkjs/remark-toc/pull/29/files#r368155887 may be another way of handling this.